java/core/src/main/java/com/whylogs/core/metrics/components/MaxIntegralComponent.java

```diff
@@ -0,0 +1,19 @@
+package com.whylogs.core.metrics.components;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper=false)
+public class MaxIntegralComponent extends IntegralComponent {
+    private final int type_id = 2;
+
+    public MaxIntegralComponent(Integer value) {
+        super(value);
+    }
+
+    // TODO: python has the @_id_aggregator decorator, how do we update?
```

Comment:
`class MetricComponentAggregator { public static <T> T aggregate(T a, T b) {raise NotImplementedException} }`
Subclass that to implement the aggregators. `MetricComponent`s can then have an aggregator member. `MetricComponent` ctor can call an aggregator factory to get a default aggregator from the registry, or subclasses can just set it.

Reply:
Makes sense. That's next on my list.